src/app/components/fairChain/fairChain.component.ts

```diff
@@ -121,18 +124,46 @@ export class FairChainComponent implements OnInit {
     } else {
       this.isAddingNodes = false;
       this.isAddingEdges = false;
-      this.network.deleteSelected();
+      this.networkDeleteSelected();
       this.isShowNodeOptions = false;
     }
   }
 
+  //Delete Nodes and Edges in selection
+  private networkDeleteSelected() {
+    let nodesToDelete: vis.IdType[] = this.network.getSelectedNodes();
+    let edgesToDelete: vis.IdType[] = this.network.getSelectedEdges();
+    this.deleteIdFromArray(this.nodes, nodesToDelete);
+    this.deleteIdFromArray(this.edges, edgesToDelete);
+    this.network.deleteSelected();
+  }
+
+  //Delete Elements that have specific id
+  private deleteIdFromArray(arrayToTrim: vis.Node[]|vis.Edge[], ids: vis.IdType[]) {
+    ids.forEach(id => {
+      let index = this.IdToIndex(arrayToTrim, id)
+      arrayToTrim.splice(index, 1);
+    });
+  }
+
+  //TODO: Change datastructure if performance is bad
+  private IdToIndex(array: vis.Node[]|vis.Edge[], id: vis.IdType) {
+    return array.findIndex(element => element.id === id);
+  }
+
   // Defines actions when user clicks on Nodes or Edges
   private onClick(params) {
     if (params.nodes && params.nodes.length >= 1) {
       if (this.isChangeNodeLabel) this.network.editNode();
     }
   }
 
+  public debugPrint() {
+    console.clear();
+    console.log(JSON.stringify(this.nodes));
+    console.log(JSON.stringify(this.edges));
+  }
+
```

Comment:
I would not push it on dev

Reply:
The overall implementation is merged in the changePageLayout